vcsclient/github.go

```diff
@@ -441,6 +444,47 @@ func (client *GitHubClient) UnlabelPullRequest(ctx context.Context, owner, repos
 	return err
 }
 
+// UploadScanningAnalysis to GitHub Security tab
+func (client *GitHubClient) UploadCodeScanning(ctx context.Context, owner, repository, branch, scan string) (string, error) {
+	packagedScan, err := packScanningResult([]byte(scan))
+	if err != nil {
+		return "", err
+	}
+	commit, err := client.GetLatestCommit(ctx, owner, repository, branch)
+	if err != nil {
+		return "", err
+	}
+	commitSHA := commit.Hash
+	ref := "refs/heads/" + branch
+	ghClient, err := client.buildGithubClient(ctx)
+	if err != nil {
+		return "", err
+	}
+	sarifID, resp, err := ghClient.CodeScanning.UploadSarif(ctx, owner, repository, &github.SarifAnalysis{
+		CommitSHA:   &commitSHA,
+		Ref:         &ref,
+		Sarif:       &packagedScan,
+		CheckoutURI: nil,
+	})
+	if err != nil && resp.Response.StatusCode != 202 {
```

Comment:
Let's just add an explanation here, regarding the 202 and the error mechanism.

Reply:
done